### PR TITLE
[feature/sidebar-help-section] Branded Sidebar Link Section

### PR DESCRIPTION
**Pull request by hosy**
## Description
Adds a Link section to the sidebar, if client is branded and all needed branding values are available
Computes a dynamic link section in the sidebar which can have multiple values.

## Related Issue
https://github.com/owncloud/enterprise/issues/5574

## How Has This Been Tested?
Use this `Branding.plist` values:

```
	<key>branding.sidebar-links</key>
	<array>
		<dict>
			<key>image</key>			
			<string>branding-sidebar-link-icon</string>
			<key>title</key>
			<string>Help Portal</string>
			<key>url</key>
			<string>https://www.owncloud.com/</string>
		</dict>
		<dict>
			<key>symbol</key>
			<string>questionmark.circle</string>
			<key>title</key>
			<string>Another Entry</string>
			<key>url</key>
			<string>https://www.owncloud.org/</string>
		</dict>
	</array>
	<key>branding.sidebar-links-title</key>
	<string>Help Section</string>
``` 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] Added an issue with details about all relevant changes in the [**iOS documentation repository**](https://github.com/owncloud/docs-client-ios-app/issues).
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

### CODE REVIEW

File: ownCloudAppShared/Branding/Branding+App.swift [hunk 1]
```diff
@@ -211,7 +214,23 @@ extension Branding : BrandingInitialization {
 				.description    : "Indicates if the user can change the server URL for the account.",
 				.status        : OCClassSettingsKeyStatus.advanced,
 				.category    : "Branding"
-			]
+			],
+            
+            .sidebarLinks : [
+                .type         : OCClassSettingsMetadataType.array,
+                .label         : "Sidebar Links",
+                .description    : "Array with Links, which should appear in the sidebar.",
```
Comment (felix-schwarz) [inline ownCloudAppShared/Branding/Branding+App.swift]: The description suggests that the parameter expects an array of URLs, whereas it expects an array of dictionaries where the keys `url` and `title` are mandatory - and where an optional image can be added as either an SF-Symbol name (`symbol`) or the name of an image bundled with the app (`image`).

**Review by felix-schwarz — CHANGES_REQUESTED** [review]

File: ownCloudAppShared/Branding/Branding+App.swift (hunk 1 above)
Comment (hosy) [reply ownCloudAppShared/Branding/Branding+App.swift]: fixed in the latest commit

**Review by felix-schwarz — APPROVED** [review]

File: ownCloudAppShared/Branding/Branding+App.swift (hunk 1 above)
Comment (felix-schwarz) [reply ownCloudAppShared/Branding/Branding+App.swift]: Thanks!

**Comment by felix-schwarz** [comment]
I just updated `experimental/12.0` to also include `feature/sidebar-help-section`.

**Comment by jesmrec** [comment]
@hosy is it posible to set different items in the list with different images? i mean, the key:

```
<key>image</key>			
<string>branding-sidebar-link-icon</string>
```

cannot be set twice into two different entries with different values, right? like:

	<array>
		<dict>
			<key>image</key>			
			<string>branding-sidebar-link-icon</string>
			<key>title</key>
			<string>Item1</string>
			<key>url</key>
			<string>https://item1.com/</string>
		</dict>
		<dict>
			<key>image</key>
			<string>branding-sidebar-link-icon</string>
			<key>title</key>
			<string>item2</string>
			<key>url</key>
			<string>https://item2.com/</string>
		</dict>
	</array>

if i'd set something different in `<key>image</key>`, image is not set. 

Any missing step?

**Pull request merged**